forum/qa-plugin/q2a-badges-master/qa-badge-check.php

```diff
@@ -358,20 +358,17 @@ function answer_vote_down($event,$event_user,$params) {
 			if($event_user) $this->check_voter($event_user);
 		}
 
-		function check_voted($uid) {
-						// upvotes received
+                function check_voted($uid) {
+                    // upvotes received
 			
-			$votes = qa_db_read_one_assoc(
-				qa_db_query_sub(
-					'SELECT upvoteds FROM ^userpoints WHERE userid=#',
-					$uid
-				),
-				true
-			);
-			$badges = array('liked','loved','revered');
+                    $votesArray = qa_db_query_raw('SELECT upvoteds FROM qa_userpoints WHERE userid='.$uid.' LIMIT 1');

```

Comment:
Nie wiem czy pozbycie się użycia funkcji `qa_db_query_sub()` rozwiązuje ten problem, ale jeśli tak to dodaje kolejne - gdy ktoś ustawi własny prefiks tabel, to to zapytanie nie zadziała. Domyślnie jest `qa_` i użycie tej funkcji na zapytaniu z `^` jak rozumiem powoduje jego dodanie. Ponadto tracimy bindowanie czy też inne zabezpieczenie zapytania, co zapewne robi ta funkcja.

Reply:
zgadzam się -> w moim odczuciu to nie rozwiązuje dobrze problemu a umożliwia wygenerowanie masy innych